app/Models/Course.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+namespace App\Models;
+
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Model;
+
+class Course extends Model
+{
+    use HasFactory;
+
+     /**
+     * The attributes that are mass assignable.
+     *
+     * @var array
+     */
+    protected $fillable = [
+        'code',
+        'name',
+    ];
+
+    public function user()
+    {
+        return $this->belongsTo( related: User::class);
+    }
```

Comment:
couse (mata kuliah) mempunyai relasi one to many terhadap  teach history (riwayat mengajar), dan tidak mempunyai relasi ke user https://drive.google.com/file/d/1ysgF9ybsiSLaV0ue1CBuUX9D_bCXrI_O/view?usp=sharing
jadi yang benar adalah
```suggestion
    public function teachHistory()
    {
        return $this->hasMany(TeachHistory::class);
    }
```